src/main/java/io/nats/client/impl/PullManagerObserver.java

```diff
@@ -15,4 +15,5 @@
 
 interface PullManagerObserver {
     void pendingUpdated();
+    void heartbeatError();
```

Comment:
needs this because the manager can't be responsible for restarts because the PMM is only managing messages, not the subscriptions / state getting messages.